contracts/strategies/BaseStrategy.sol

```diff
@@ -0,0 +1,217 @@
+                    return diff;
+                } else {
+                    // we made a loss but we have some tokens we can reinvest
+                    _skim(contractBalance);
+                    return diff;
+                }
+            } else {
+                // we made a loss
+                return amount;
+            }
+        }
+        return int256(0);
+    }
+
+    /// @notice Harvests any rewards and rebalances them to the underlying token
+    /// @dev Doesn't need to be overriden if we don't expect any rewards.
+    function _harvestRewards() internal virtual {}
+
+    /// @notice Harvest any profits made and transfer them to address(this) or report a loss
+    /// @param balance The amount of tokens that have been invested.
+    /// @return amountAdded The delta (+profit or -loss) that occured in contrast to `balance`.
+    /// @dev amountAdded can be left at 0 when reporting profits (gas savings).
+    function _harvest(uint256 balance) internal virtual returns (int256 amountAdded);
+
+    /// @inheritdoc IStrategy
+    function withdraw(uint256 amount) external override onlyBentobox returns (uint256 actualAmount) {
+        _withdraw(amount);
+        /// @dev Make sure we send and report the exact same amount of tokens by using balanceOf.
+        actualAmount = IERC20(underlying).safeBalanceOf(address(this));
+        IERC20(underlying).safeTransfer(address(bentoBox), actualAmount);
+    }
+
+    /// @dev Withdraw the requested amount of the underlying tokens to address(this)
+    /// @param amount The requested amount the caller wants to withdraw.
+    function _withdraw(uint256 amount) internal virtual;
+
+    /// @inheritdoc IStrategy
+    function exit(uint256 balance) external override onlyBentobox returns (int256 amountAdded) {
+        _exit();
+        /// @dev Check balance of token on the contract.
+        uint256 actualBalance = IERC20(underlying).safeBalanceOf(address(this));
+        /// @dev Calculate tokens added (or lost).
+        amountAdded = int256(actualBalance) - int256(balance);
+        /// @dev Transfer all tokens to bentoBox.
+        underlying.safeTransfer(address(bentoBox), actualBalance);
+        /// @dev Flag as exited, allowing the owner to manually deal with any amounts available later.
+        exited = true;
+    }
+
+    /// @notice Withdraw the maximum amount of the invested assets to address(this).
+    /// @dev This shouldn't revert (use try catch).
+    function _exit() internal virtual;
+
+    function afterExit(
+        address to,
+        uint256 value,
+        bytes memory data
+    ) public onlyOwner returns (bool success) {
+        /** @dev After exited, the owner can perform ANY call. This is to rescue any funds that didn't
+			get released during exit or got earned afterwards due to vesting or airdrops, etc. */
+        require(exited, "BentoBox Strategy: not exited");
+        (success, ) = to.call{value: value}(data);
+    }
+
+    // **** SWAP ****
```

Comment:
Do we want to include the current swap architecture even if we are moving on soon enough?